hooks/useAsyncLoad.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export default function useAsyncLoad<T>(fn: () => Promise<T>) {
+  const funcRef = useRef<typeof fn>(fn);
+  const [data, setData] = useState<T>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    funcRef.current = fn;
+  }, [fn]);
```

Comment:
Every time `fn` changes, the value of `funcRef.current` also changes because of the `useEffect`. So `funcRef.current` always reflects the most recent value of `fn`. I believe you could use `fn` directly instead of maintaining and updating a separate reference.